docs/sphinx/specification/cudaq/synthesis.rst

```diff
@@ -1,67 +1,106 @@
-        cudaq::qarray<3> q;
-        ...
-        // Create Toffoli gate
-        auto ctrl_bits = q.front(2);
-        cudaq::control(x_gate{}, ctrl_bits, q[2]);
-        ...
-      }
-    };
-    struct rx_and_h_gate {
-      void operator()(double x, cudaq::qubit& q) __qpu__ { rx(x,q); h(q); }
-    };
-    struct kernel {
-      void operator(int N) () __qpu__ {
+  .. code-block:: cpp
+
+      __qpu__ void x_gate(cudaq::qubit& q) { x(q); }
+      
+      struct kernel {
+        void operator() () __qpu__ {
+          cudaq::qarray<3> q;
+          ...
+          // Create Toffoli gate
+          auto ctrl_bits = q.front(2);
+          cudaq::control(x_gate, ctrl_bits, q[2]);
+          ...
+        }
+      };
+      
+      void rx_and_h_gate(double x, cudaq::qubit& q) __qpu__ { rx(x,q); h(q); }
+      
+      __qpu__ kernel(int N) {
         cudaq::qvector q(N);
         ...
         // apply h(q[2]); rx(-pi, q[2]);
         cudaq::adjoint(rx_and_h_gate{}, M_PI, q[2]);
         ...
       }
-    };
 
-The :code:`cudaq::control(...)` function takes as input an instantiated pure
+.. tab:: Python 
+
+  .. code-block:: python 
+
+    @cudaq.kernel()
+    def x_gate(q : cudaq.qubit):
+        x(q)
+    
+    @cudaq.kernel()
+    def kernelTestControl():
+        q = cudaq.qvector(3)
+        ...
+        ctrl_bits = q.front(2)
+        cudaq.control(x_gate, ctrl_bits, q[2])
+        ...
+    
+    @cudaq.kernel()
+    def rx_and_h_gate(x : float, q : cudaq.qubit):
+        rx(x, q)
+        h(q)
+    
+    @cudaq.kernel()
+    def kernelTestAdjoint(N : int):
+        q = cudaq.qvector(N)
+        ...
+        # apply h(q[2]); rx(-pi, q[2])
+        cudaq.adjoint(rx_and_h_gate, np.pi, q[2])
+        ...
+
+
+**[3]** The :code:`cudaq::control(...)` function takes as input an instantiated pure
 device quantum kernel, a std::range of control qubits (:code:`cudaq::qvector`
 or :code:`cudaq::qview`), and the remaining arguments for the kernel itself.
-Compiler implementations are free to synthesize multi-controlled operations
+
+**[4]** Compiler implementations are free to synthesize multi-controlled operations
 using any pertinent synthesis strategy available. Qubits may be aggregated into
 a range of control qubits with or without the use of the :code:`operator!`
 :doc:`negated polarity operator <operations>`.
 
-.. code-block:: cpp
+.. tab:: C++ 
+  
+  .. code-block:: cpp
+
+      cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
+
+.. tab:: Python 
 
-    cudaq::control(kernel{}, {qubit0, !qubit1}, kernel_arg);
+  .. code-block:: python 
 
-The :code:`cudaq::adjoint(...)` function takes as input an
-instantiated pure device quantum kernel (or specified template type)
-and the remaining arguments for the kernel.
+    cudaq.control(kernel, [qubit0, ~qubit1], kernel_arg)
+  
+**[5]** The :code:`cudaq::adjoint(...)` function takes as input an
+pure device quantum kernel and the remaining arguments for the kernel.
```

Comment:
Q: Should this include `cudaq.synthesize()`?
Defined here - https://github.com/NVIDIA/cuda-quantum/blob/main/python/cudaq/__init__.py#L81

Reply:
Also, what about `cudaq.to_qir()`?

Reply:
At this point no - those functions are not part of the specification. Maybe we can add them in the future